src/SourceGenerators/Uno.UI.SourceGenerators/TSBindings/TSBindingsGenerator.cs

```diff
@@ -0,0 +1,344 @@
+				_floatSymbol = context.Compilation.GetTypeByMetadataName("System.Single");
+				_doubleSymbol = context.Compilation.GetTypeByMetadataName("System.Double");
+				_byteSymbol = context.Compilation.GetTypeByMetadataName("System.Byte");
+				_shortSymbol = context.Compilation.GetTypeByMetadataName("System.Int16");
+				_intPtrSymbol = context.Compilation.GetTypeByMetadataName("System.IntPtr");
+				_boolSymbol = context.Compilation.GetTypeByMetadataName("System.Boolean");
+				_longSymbol = context.Compilation.GetTypeByMetadataName("System.Int64");
+				_structLayoutSymbol = context.Compilation.GetTypeByMetadataName(typeof(System.Runtime.InteropServices.StructLayoutAttribute).FullName);
+
+				GenerateTSMarshallingLayouts();
+			}
+		}
+
+		internal void GenerateTSMarshallingLayouts()
+		{
+			foreach (var parametersType in _windowManagerSymbol.GetTypeMembers().Where(t => t.IsValueType))
+			{
+				var packValue = GetStructPack(parametersType);
+
+				var sb = new IndentedStringBuilder();
+
+				sb.AppendLineInvariant($"/* {nameof(TSBindingsGenerator)} Generated code -- this code is regenerated on each build */");
+
+				using (sb.BlockInvariant($"class {parametersType.Name}"))
+				{
+					sb.AppendLineInvariant($"/* Pack={packValue} */");
+
+					foreach (var field in parametersType.GetFields())
+					{
+						sb.AppendLineInvariant($"{field.Name} : {GetTSFieldType(field.Type)};");
+					}
+
+					if (parametersType.Name.EndsWith("Params"))
+					{
+						GenerateSerializer(parametersType, sb, packValue);
+					}
+
+					if (parametersType.Name.EndsWith("Return"))
+					{
+						GenerateDeserializer(parametersType, sb, packValue);
+					}
+				}
+
+				var outputPath = Path.Combine(_bindingsPaths, $"{parametersType.Name}.ts");
+
+				File.WriteAllText(outputPath, sb.ToString());
+			}
+		}
+
+		private int GetStructPack(INamedTypeSymbol parametersType)
+		{
+			// https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/Portable/Symbols/TypeLayout.cs is not available.
+
+			if(parametersType.GetType().GetProperty("Layout", System.Reflection.BindingFlags.Instance|System.Reflection.BindingFlags.NonPublic) is PropertyInfo info)
+			{
+				if(info.GetValue(parametersType) is object typeLayout)
+				{
+					if (typeLayout.GetType().GetProperty("Kind", BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic) is PropertyInfo layoutKingProperty)
+					{
+						if (((System.Runtime.InteropServices.LayoutKind)layoutKingProperty.GetValue(typeLayout)) == System.Runtime.InteropServices.LayoutKind.Sequential)
+						{
+							if (typeLayout.GetType().GetProperty("Alignment", BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.Public) is PropertyInfo alignmentProperty)
+							{
+								return (short)alignmentProperty.GetValue(typeLayout);
+							}
+						}
+						else
+						{
+							throw new InvalidOperationException($"The LayoutKind for {parametersType} must be LayoutKind.Explicit");
```

Comment:
You checked for `Sequential`
```suggestion
							throw new InvalidOperationException($"The LayoutKind for {parametersType} must be LayoutKind.Sequential");
```